boa/src/builtins/array/mod.rs

```diff
@@ -286,6 +287,32 @@ impl Array {
         Ok(array_obj_ptr)
     }
 
+    /// Utility function for concatenating array objects.
+    ///
+    /// Returns a Boolean valued property that if `true` indicates that
+    /// an object should be flattened to its array elements
+    /// by `Array.prototype.concat`.
+    fn is_concat_spreadable(this: &GcObject, context: &mut Context) -> Result<bool> {
+        // 1. If Type(O) is not Object, return false.
```

Comment:
But it now needs to accept &GcObject for spec compliance right?

Reply:
> But it now needs to accept &GcObject for spec compliance right?

In the spec it says:
```
1. If Type(O) is not Object, return false.
2. Let spreadable be ? Get(O, @@isConcatSpreadable).
3. If spreadable is not undefined, return ! ToBoolean(spreadable).
4. Return ? IsArray(O).
```

It does not say `1. Assert: Type(O) is Object.` which means it always is an object, rather it says `1. If Type(O) is not Object, return false` Which means it could a value which is not an object, in that case return false. So the previous implementation was more spec compliant.

Reply:
Yep, that's my bad... Sorry about that!